### PR TITLE
Add Google Analytics metrics for auto-renewing loans

### PR DESCRIPTION
**Pull request by nsharma123**
https://webarchive.jira.com/browse/WEBDEV-5909

### CODE REVIEW

**Review by iisa — COMMENTED** [review]
@nsharma123 can you give a few lines on your approach for the analytics piece? the changes touch many files... thanks in advance.

File: src/ia-book-actions.js [hunk 1]
```diff
@@ -410,6 +405,9 @@ export default class IABookActions extends LitElement {
     await this.localCache.delete(`${this.identifier}-loanTime`);
     await this.localCache.delete(`${this.identifier}-pageChangedTime`);
 
+    // remove loan-renew-count-<identifier> cookie
+    Cookies.removeItem(`loan-renew-count-${this.identifier}`);
```
Comment (nsharma123) [inline src/ia-book-actions.js:409]: need to cookie change name, Forgot!

File: src/core/services/loan-token-poller.js [hunk 2]
```diff
@@ -65,7 +72,11 @@ export class LoanTokenPoller {
         this.borrowType === 'browsed' ? 'BrowseCounts-' : 'Counts-'
       }${consecutiveLoanCounts}`;
 
-      this.sendEvent(category, action);
+      this.loanAnalytics?.sendEvent(
+        category,
+        action,
+        `identifier=${this.identifier}`
```
Comment (nsharma123) [inline src/core/services/loan-token-poller.js:78]: this.identifer vs `identifier=${this.identifier}???

File: src/core/services/loan-analytics.js [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+        this.gaStats.renew = renew;
+        break;
+
+      case 'return' || 'autoreturn':
+        expire = expire ? Number(expire) + 1 : 1;
+        this.gaStats.expire = expire;
+
+        // reset renew and expire count when browse
+        renew = 0;
+        expire = 0;
```
Comment (nsharma123) [inline src/core/services/loan-analytics.js:97]: remvoe this..!

File: src/core/services/loan-analytics.js [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+
+    // store these counts in cookies
+    this.lendingEventCounts = { browse, renew, expire };
+
+    // obtain GA category, event
+    // const { category, event } = await this.getGACategoryAction(action);
+    const category = analyticsCategories.browse;
+    const event = `browse${this.paddedNumber(
+      this.gaStats?.browse
+    )}-autorenew${this.paddedNumber(this.gaStats?.renew)}:${action}`;
```
Comment (nsharma123) [inline src/core/services/loan-analytics.js:111]: autorenew:browse001-autorenew010 ???

File: src/core/services/loan-analytics.js [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+   *
+   *
+   * @param {string} action - like 'browse|autorenew|etc...'
+   * @return {object} - contains GA category and event
+   * - @param {string}
+   * - @param {string}
+   *
+   * @memberof LoanAnanlytics
+   */
+  getGACategoryAction(action) {
```
Comment (nsharma123) [inline src/core/services/loan-analytics.js:131]: can be removed.!

File: src/core/services/loan-analytics.js [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+   * reponsible to sent events to GA
+   *
+   * @param {*} eventCategory
+   * @param {*} eventAction
+   * @param {*} label
+   * @param {*} extraParams
+   * @memberof LoanAnanlytics
+   */
+  sendEvent(eventCategory, eventAction, label, extraParams) {
+    console.debug(
```
Comment (nsharma123) [inline src/core/services/loan-analytics.js:171]: I don't know.

**Review by iisa — APPROVED** [review]
cr 1:1 in person.
approach looks good. lots of files touched, but made for simple organization. 👍 
looks good to merge after he changes we talked about are in. solid work!

**Comment by iisa** [comment]
neeraj please bump up tag to next patch value after merge, thanks!

File: src/core/services/loan-analytics.js (hunk 6 above)
Comment (iisa) [reply src/core/services/loan-analytics.js:171]: we will keep

**Comment by iisa** [comment]
@nsharma123 please merge in https://github.com/internetarchive/iaux-book-actions/pull/59 into this PR before merging to main line. i had to add some logging and found a place where the toast was being suppressed. thanks in advance

**Pull request merged**